### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by alex-galych**
- Developer(s): @vhrychenko

- Peer Reviewer: 

- Tickets:
        https://spryker.atlassian.net/browse/ECO-1686

- Academy PR: 

#### Please confirm

- [x] No new OS components - or they have been approved by the legal department
- [x] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [x] Functional documentation provided or in progress?
- [x] Integration guide for projects provided or not needed?
- [x] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraints         |
   :--------------------- | :------------------------ | :--------------------- |
   Inxmail               |       major                 |                       |

#### Release Notes
 - 1.0.0 version for inxmail module.

-----------------------------------------

#### Module Inxmail

_**Patch:** Backwards-compatible bug fix_

Def of done (by the responsible developer):
- [x] All changes are backwards-compatible. The outdated code is marked as deprecated.
- [x] The change is isolated and not mixed with any cleanups or other changes.
- [x] New code fits the architecture rules.
- [x] All new or changed business logic is covered by functional tests (in Zed business layer).

###
- 1.0 version for inxmail module.

### CODE REVIEW

File: .scrutinizer.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+build:
+    dependencies:
+        before:
+            - composer config repositories.spryker composer https://code.spryker.com/repo/private
```
Comment (dereuromark) [inline .scrutinizer.yml]: We dont use this anymore since 2017.

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    "spryker/mail": "^4.0.0"
+  },
+  "require-dev": {
+    "spryker/code-sniffer": "*",
+    "spryker/monolog": "*",
+    "spryker/queue": "*",
+    "spryker/testify": "*"
+  },
+  "autoload": {
+    "psr-0": {
```
Comment (dereuromark) [inline composer.json]: we should use psr-4 always.

File: src/SprykerEco/Shared/Inxmail/InxmailConstants.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public const KEY_ID = 'INXMAIL:KEY_ID';
+    public const SECRET = 'INXMAIL:SECRET';
+
+    public const API_EVENT_URL = 'INXMAIL:API_EVENT_URL';
+
+    public const EVENT_CUSTOMER_REGISTRATION = 'INXMAIL:EVENT_CUSTOMER_REGISTRATION';
+    public const EVENT_CUSTOMER_RESET_PASSWORD = 'INXMAIL:EVENT_CUSTOMER_RESET_PASSWORD';
+    public const EVENT_ORDER_NEW = 'INXMAIL:EVENT_ORDER_NEW';
+    public const EVENT_ORDER_SHIPPING_CONFIRMATION = 'INXMAIL:EVENT_ORDER_SHIPPING_CONFIRMATION';
+    public const EVENT_ORDER_CANCELED = 'INXMAIL:EVENT_ORDER_CANCELED';
```
Comment (dereuromark) [inline src/SprykerEco/Shared/Inxmail/InxmailConstants.php]: Are you sure those are valid ENV configs as outlined in the class doc block above and per guidelines?
Also, cancelled has to L.

File: src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+     * @param \ArrayObject $methods
+     *
+     * @return string
+     */
+    protected function getPaymentMethodsTotal(ArrayObject $methods): string
+    {
+        $sum = 0;
+
+        /**
+         * @var \Generated\Shared\Transfer\PaymentTransfer $method
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php:247]: Dont do this, instead annotate the incoming param as `|...[]`

File: composer.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
     "spryker/testify": "*"
   },
   "autoload": {
-    "psr-0": {
+    "psr-4": {
       "SprykerEco": "src/",
```
Comment (dereuromark) [inline composer.json]: Invalid psr-4 here, please see other code how it works.

Comment (vhrychenko) [reply composer.json]: Yes, I've already seen, Thanks

File: .scrutinizer.yml (hunk 1 above)
Comment (vhrychenko) [reply .scrutinizer.yml]: Removed

File: composer.json (hunk 2 above)
Comment (vhrychenko) [reply composer.json]: Updated

File: src/SprykerEco/Shared/Inxmail/InxmailConstants.php (hunk 3 above)
Comment (vhrychenko) [reply src/SprykerEco/Shared/Inxmail/InxmailConstants.php]: L is updated. Event configuration is set on the Inxmail side and events can be different for env to env.

File: src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php [hunk 6]
```diff
@@ -215,7 +215,7 @@ protected function getOrderDeliveryInfo(ArrayObject $methods, ArrayObject $expen
     }
 
     /**
-     * @param \ArrayObject|\Generated\Shared\Transfer\PaymentTransfer $methods
+     * @param \ArrayObject $methods
```
Comment (dereuromark) [inline src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php]: \Generated\Shared\Transfer\PaymentTransfer[] !

Comment (dereuromark) [reply src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php]: Doesnt this fail without the proper typehint documentation here in phpstan?

**Pull request merged**